src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -190,15 +191,32 @@ public async Task<T> GetConfigurationAsync(CancellationToken cancel)
                     catch (Exception ex)
                     {
                         _fetchMetadataFailure = ex;
-                        _syncAfter = DateTimeUtil.Add(now.UtcDateTime, AutomaticRefreshInterval < RefreshInterval ? AutomaticRefreshInterval : RefreshInterval);
+
                         if (_currentConfiguration == null) // Throw an exception if there's no configuration to return.
+                        {
+                            if (_bootstrapRefreshInterval < DefaultRefreshInterval)
+                            {
+                                // Adopt exponential backoff for bootstrap refresh interval with a jitter if it is not longer than 5 minutes.
+                                _bootstrapRefreshInterval += _bootstrapRefreshInterval;
```

Comment:
>_bootstrapRefreshInterval [](http://example.com/codeflow?start=32&length=25)

This value is never reset, we should set it back to the default value upon successful retrieval of the config.

Reply:
I think we only bootstrap once. Once we get a config we use that stale config forever and only retry every `RefreshInterval`